### PR TITLE
#233 - #234 - meta tags and facebook meta tags

### LINKED ISSUE #233: [SEO] Meta tags
All pages - dynamic and static should have [meta tags](https://widoczni.com/blog/meta-tagi-w-seo/) dedicated for Google:
- title
- description
- robots

### LINKED ISSUE #234: [SEO] Facebook meta tags
All pages - dynamic and static should have meta tags dedicated for Facebook [og:tags](https://developers.facebook.com/docs/sharing/webmasters/):
- title
- description
- image
- url
- type
- app_id (optional, if we want connect that with facebook analytics stuff)

### PR DESCRIPTION
**Pull request by AleksandraKozubal**
Added simple meta tags for pages and facebook meta tags for activities.

It should close #233 and close #234.

### CODE REVIEW

File: resources/views/layout/public.blade.php [hunk 1]
```diff
@@ -2,9 +2,11 @@
 <html lang="{{ App::currentLocale() }}">
 
 <head>
-    <title>Blumilk</title>
+    <title>Blumilk - @yield('title')</title>
```
Comment (kamilpiech97) [inline resources/views/layout/public.blade.php:5]:
```suggestion
    <title>@yield('title') - Blumilk</title>
```

File: resources/views/activity.blade.php [hunk 2]
```diff
@@ -1,6 +1,15 @@
 @extends("layout.public")
 
+@section("title", $activity->title)
+@section("description", $activity->subtitle)
+
 @section("content")
+    <meta property="og:url"                content="https://blumilk.pl/pl/aktywnosci/{{ $activity->slug }}" />
```
Comment (kamilpiech97) [inline resources/views/activity.blade.php:7]: what about `en`?

File: resources/views/activities.blade.php [hunk 3]
```diff
@@ -1,5 +1,8 @@
 @extends("layout.public")
 
+@section("title", __("meta.activities.title"))
+@section("description", __("meta.activities.description"))
```
Comment (kamilpiech97) [inline resources/views/activities.blade.php:4]: Are u sure? how long subtitle can be?
Maybe separated fields for SEO would be better?

File: resources/views/activity.blade.php (hunk 2 above)
Comment (krzysztofrewak) [reply resources/views/activity.blade.php:7]: And what about these multiplicated spaces between attributes?

Comment (AleksandraKozubal) [reply resources/views/activity.blade.php]: Added universal route, removed spaces

File: resources/views/activities.blade.php (hunk 3 above)
Comment (AleksandraKozubal) [reply resources/views/activities.blade.php:4]: We're talking about activity fb tags - they say usually between 2 and 4 sentences. Currently subtitle max length is 255 characters, so it matches.

It doesn't matter whether these fields are used for SEO or for an article - in both cases we should keep them short.

File: resources/views/activity.blade.php [hunk 4]
```diff
@@ -1,6 +1,20 @@
 @extends("layout.public")
 
+@section("title", $activity->title)
+@section("description", $activity->subtitle)
+
+@php
+    $urlPath = request()->getRequestUri();
+    $articleUrl = "https://blumilk.pl".$urlPath;
+@endphp
```
Comment (kamilpiech97) [inline resources/views/activity.blade.php:9]: I would like to move that into controller.

File: app/Http/Controllers/ActivitiesController.php [hunk 5]
```diff
@@ -39,8 +39,11 @@ public function index(Request $request, Factory $factory): View
             ->with("selectedTag", $tag?->title);
     }
 
-    public function get(Factory $factory, string $slug): View
+    public function get(Request $request, Factory $factory, string $slug): View
     {
+        $urlPath = $request->getRequestUri();
+        $articleUrl = "https://blumilk.pl" . $urlPath;
```
Comment (krzysztofrewak) [inline app/Http/Controllers/ActivitiesController.php:45]: I don't get why URL is hardcoded here. We have `config("app.url")` value. But we can also add `/whatever` and browser should add main domain automatically.

Comment (AleksandraKozubal) [reply app/Http/Controllers/ActivitiesController.php]: Added config here

File: lang/pl/meta.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+declare(strict_types=1);
+
+return [
+    "home" => [
+        "title" => "Strona główna",
+        "description" => "Oprogramowanie dostosowane do Twoich potrzeb. Zobacz, jak możemy Ci pomóc!",
+    ],
+    "about" => [
+        "title" => "O firmie",
+        "description" => "Jesteśmy pasjonatami tworzenia aplikacji web/mobile oraz zaawansowanych projektów B+R.. Zmieniamy koncepcje w działające rozwiązania.",
```
Comment (EwelinaSkrzypacz) [inline lang/pl/meta.php:12]:
```suggestion
        "description" => "Jesteśmy pasjonatami tworzenia aplikacji web/mobile oraz zaawansowanych projektów B+R. Zmieniamy koncepcje w działające rozwiązania.",
```

File: lang/pl/meta.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        "title" => "Kariera",
+        "description" => "Dołącz do naszego zespołu!",
+    ],
+    "software-engineer" => [
+        "title" => "Inżynier oprogramowania",
+        "description" => "Dołącz do naszego zespołu jako inżynier oprogramowania!",
+    ],
+    "contact" => [
+        "title" => "Kontakt",
+        "description" => "Jeśli szukasz rozwiązania dostosowanego do Twoich unikalnych potrzeb, jesteśmy dla Ciebie! Czekamy na Twoją wiadomość!",
```
Comment (EwelinaSkrzypacz) [inline lang/pl/meta.php:32]:
```suggestion
        "description" => "Jeśli szukasz rozwiązania idealnie dopasowanego do Twoich potrzeb, jesteśmy dla Ciebie! Czekamy na Twoją wiadomość!",
```

File: lang/en/meta.php [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        "title" => "Career",
+        "description" => "Join our team!",
+    ],
+    "software-engineer" => [
+        "title" => "Software engineer",
+        "description" => "Join our team as a software engineer!",
+    ],
+    "contact" => [
+        "title" => "Contact",
+        "description" => "If you are looking for a solution tailored to your unique needs, we are here for you! We are waiting for your message!",
```
Comment (EwelinaSkrzypacz) [inline lang/en/meta.php:32]:
```suggestion
        "description" => "If you are looking for a solution perfectly tailored to your needs, we are here for you! We are waiting for your message!",
```

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**